api_app/analyzers_manager/observable_analyzers/misp.py

```diff
@@ -25,21 +29,19 @@ def run(self):
             debug=self.debug,
             timeout=5,
         )
-
-        # we check only for events not older than 90 days and max 50 results
         now = datetime.datetime.now()
-        date_from = now - datetime.timedelta(days=90)
+        date_from = now - datetime.timedelta(days=self.from_days)
         params = {
-            # even if docs say to use "values",...
-            # .. at the moment it works correctly only with "value"
             "value": self.observable_name,
-            "type_attribute": [self.observable_classification],
-            "date_from": date_from.strftime("%Y-%m-%d %H:%M:%S"),
-            "limit": 50,
-            "enforce_warninglist": True,
+            "limit": self.limit,
+            "enforce_warninglist": self.enforce_warninglist,
         }
-        if self.observable_classification == self.ObservableTypes.HASH:
-            params["type_attribute"] = ["md5", "sha1", "sha256"]
+        if self.from_days != 0:
+            params["date_from"] = date_from.strftime("%Y-%m-%d %H:%M:%S")
+        if self.filter_on_type:
```

Comment:
considering that we analyze a single observable at a time, there is no need for this flag.

The rest is good! thank you!

Reply:
I'll merge after this change

Reply:
Which flag are you talking about ?

Reply:
`if self.filter_on_type:`

Reply:
I have to admit that I don't get your point. 
The goal of this flag is to give option for the user to search (or not) on the attribute type in their MISP instance.
In my case my MISP instance doesn't specify the type of observable for entries, thus if I specify the type_attribute in the MISP query it return no queries. 
If I am wrong could you explain broadly why the flag is useless? Thanks

Reply:
First, I have not been working with a MISP instance for quite some time, so thanks for helping me understanding the reason of this change.

The reason why, right now, the parameter `type_attribute` is mandatory is:
* all the attributes must have a category: url, domain, etc. So are you telling me that now this is no more mandatory? Can you ping me either documentation or a real use case where this actually happens?
* performance. This filter reduces the time required to perform the query. This is critical for scalability purposes. This is why I cannot easily accept this change without being sure 100% that we are doing the right thing.

Reply:
## Mandatory 
[Here ](https://buildmedia.readthedocs.org/media/pdf/pymisp/latest/pymisp.pdf) (page 29) is the doc specifying than it is optional : 
`type_attribute (Optional[TypeVar(SearchParameterTypes, str,
List[Union[str, int]], Dict[str, Union[str, int]])]) – The attribute type, any
valid MISP attribute type is accepted.`

"Any valid MISP attribute type is accepted", [here](https://www.misp-project.org/datamodels/) are the misp attribute type and "ip" isn't an existing one. 

### Filter on type
No entry returned
![image](https://user-images.githubusercontent.com/103488825/163575020-6fcc381f-a9bd-4f10-92f1-8553eb395e3f.png)

### No filter on type 
One entry returned 
#### Pic
![image](https://user-images.githubusercontent.com/103488825/163575280-09348cf8-4705-4fff-9e8f-5608fdd55f11.png)

### Remark
As we can see in last picture the type is in my case "ip-dst", thus I believe than intelowl filter on "ip" and is a strict search, I can investigate on the use of wildcard for "ip*" if you want, or use each attribute type for IPs.  

## Performance
I got the point but by default the parameter is set to true. It wouldn't change anything to common user. (unless an if comparison which is negligible)

Reply:
thank you again for the time invested in this! all extremely useful.

Ok, so at the end the big problem is for the IP addresses that were *never* extracted. This is a pretty bad bug. I'll definitevely make a release out of that.

So I'd say that we could leave the code like you made it + just customize the `type_attribute` list in case the observable is an IP address.

```
            if self.observable_classification == self.ObservableTypes.IP:
                params["type_attribute"] = ["ip-dst", "ip-src", "ip-src|port", "ip-dst|port"]
```

What do you think?

Reply:
> Ok, so at the end the big problem is for the IP addresses that were _never_ extracted. 
At least not on this version of misp. 

> I'd say that we could leave the code like you made it
IMHO the more configurable it is for the end user the better

> just customize the type_attribute 
Yes totally, concerning the type _domain|ip_ would it go for ip, domain, both or none ? I would say both

Reply:
yes, I'd say both too. better adding than removing in any case I think.

Reply:
Ok I'll handle that

Reply:
Done but I messed up with the rebase... Hope you can catch up with my mistake, otherwise I have a clean branch that didn't rebase your commit, I can make a new PR if it is better for you !

Reply:
np I will squash the commits here anyway so the merge should be clean